processing/src/main/java/io/druid/query/groupby/strategy/GroupByStrategyV2.java

```diff
@@ -131,7 +139,8 @@ public static DateTime getUniversalTimestamp(final GroupByQuery query)
   public GroupByQueryResource prepareResource(GroupByQuery query, boolean willMergeRunners)
   {
     if (!willMergeRunners) {
-      final int requiredMergeBufferNum = countRequiredMergeBufferNum(query, 1);
+      final int requiredMergeBufferNum = countRequiredMergeBufferNum(query, 1) +
+                                         (query.getSubtotalsSpec() != null ? 1 : 0);
```

Comment:
Do we really need an extra merge buffer when we’re computing subtotals? There’s already a requirement that the subtitals dimensions be in the same order as the top level dimensions, meaning we should be able to compute them without a big extra buffer. Just one row of scratch space plus a streaming combine.

Reply:
Oh wait, I'm dumb, this isn't true. If we did a group by on A, B, C and wanted an A, C subtotal, then we'll be seeing values of C non-contiguously. Nevermind!

Reply:
It would be nice in the future to optimize for the case where all subtotals _can_ be done streaming (if they are all prefixes) but that could be future work, not in this PR.

Reply:
yeah, that optimization would be nice.